internal/framework/subproviders/morpheus/resources/datastore/datastore_read.go

```diff
@@ -89,8 +89,44 @@ func getDatastoreAsState(
 	switch *refType {
 	case cloudRefType:
 		state.AssociatedResourceType = types.StringValue(associatedResourceTypeCloud)
+		// For imports, we need to get the tenants and resource permissions from the cloud datastore API
+		if plan.Name.IsNull() || plan.Name.IsUnknown() {
+			tenants, resourcePermissions, pdiags := populateCloudDatastoreInformation(
+				ctx, id, *datastore.RefId, client)
+			diags = append(diags, pdiags...)
+			if diags.HasError() {
+				return state, diags
+			}
+			state.Tenants = tenants
+			state.ResourcePermissions = resourcePermissions
+
+			break
+		}
+
+		// If not importing, set from plan
+		state.ResourcePermissions = plan.ResourcePermissions
+		state.Tenants = plan.Tenants
+
 	case clusterRefType:
 		state.AssociatedResourceType = types.StringValue(associatedResourceTypeCluster)
+		// For imports, we need to get the tenants and resource permissions from the cluster datastore API
+		if plan.Name.IsNull() || plan.Name.IsUnknown() {
+			tenants, resourcePermissions, pdiags := populateClusterDatastoreInformation(
+				ctx, id, *datastore.RefId, client)
+			diags = append(diags, pdiags...)
+			if diags.HasError() {
```

Comment:
Checking `diags.HasError()` but populate functions only add warnings. I'm guessing this error check wouldn't trigger